test/integration/src/test/java/io/pravega/test/integration/CheckpointTest.java

```diff
@@ -167,6 +173,471 @@ public void testCheckpointAndRestore() throws ReinitializationRequiredException,
+        CompletableFuture<Checkpoint> checkpoint = readerGroup.initiateCheckpoint("Checkpoint", backgroundExecutor);
+        assertFalse(checkpoint.isDone());
+        read = reader.readNextEvent(60000);
+        assertTrue(read.isCheckpoint());
+        assertEquals("Checkpoint", read.getCheckpointName());
+        assertNull(read.getEvent());
+
+        //Read 2 more event ie event 2 and event 3
+        assertEquals(testString2, reader.readNextEvent(1000).getEvent());
+        assertEquals(testString3, reader.readNextEvent(1000).getEvent());
+
+        //Modify RG to add new stream s2 and start reading from stream s1 (streamcut sc1). SC1 positions are greater than checkpoint.
+        clock.addAndGet(CLOCK_ADVANCE_INTERVAL);
+        readerGroup.resetReaderGroup(groupConfigSC);
+        try {
+            reader.readNextEvent(60000);
+            fail();
+        } catch (ReinitializationRequiredException e) {
+            //Expected
+        }
+        reader.close();
+        reader = clientFactory.createReader(readerName, readerGroupName, serializer, ReaderConfig.builder().build());
+
+        //reset to start of streamcut of S1 and new stream s2
+        clock.addAndGet(CLOCK_ADVANCE_INTERVAL);
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+
+        //Reset reader group to null been called which essentially reset readergroup to streamcut sc1 of stream s1 and head of stream s2.
+        clock.addAndGet(CLOCK_ADVANCE_INTERVAL);
+        readerGroup.resetReaderGroup(null);
+        try {
+            reader.readNextEvent(60000);
+
+        } catch (ReinitializationRequiredException e) {
+            //Expected
+        }
+        reader.close();
+        reader = clientFactory.createReader(readerName, readerGroupName, serializer, ReaderConfig.builder().build());
+
+        //reset to start of streamcut of S1(event 4 onwards in total 3 event from stream s1) and new stream s2(7 events)
+        clock.addAndGet(CLOCK_ADVANCE_INTERVAL);
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNull(reader.readNextEvent(1000).getEvent());
```

Comment:
Here and above, wherever possible pass `0` for the `readNextEvent` timeout. This guarantees that the test actually works without any dependency on time. (The reader should always make at least one attempt to read even with a zero timeout)

Reply:
Modified accordingly.

Reply:
As this testcase is part of Integration test added timeout for readNextEvent call to fetch event from the segments.